L1Trigger/L1TTrackMatch/plugins/L1TkMuonProducer.cc

```diff
@@ -665,36 +650,7 @@ void L1TkMuonProducer::build_tkMuons_from_idxs(TkMuonCollection& tkMuons,
                                                const edm::Handle<RegionalMuonCandBxCollection>& muonH,
                                                int detector) const {
   for (uint imatch = 0; imatch < matches.size(); ++imatch) {
-    int match_trk_idx = matches.at(imatch);
-    if (match_trk_idx < 0)
-      continue;  // this muon was not matched to any candidate
-
-    // take properties of the track
-    const L1TTTrackType& matchTk = (*l1tksH.product())[match_trk_idx];
-    const auto& p3 = matchTk.momentum();
-    const auto& tkv3 = matchTk.POCA();
-    float p4e = sqrt(mu_mass * mu_mass + p3.mag2());
-    math::XYZTLorentzVector l1tkp4(p3.x(), p3.y(), p3.z(), p4e);
-
-    edm::Ptr<L1TTTrackType> l1tkPtr(l1tksH, match_trk_idx);
-    edm::Ref<RegionalMuonCandBxCollection> l1muRef(muonH, imatch);
-
-    float trkisol = -999;
-    TkMuon l1tkmu(l1tkp4, l1muRef, l1tkPtr, trkisol);
-    l1tkmu.setTrackCurvature(matchTk.rInv());
-    l1tkmu.setTrkzVtx((float)tkv3.z());
-    l1tkmu.setMuonDetector(detector);
-    tkMuons.push_back(l1tkmu);
-  }
-  return;
-}
-
-void L1TkMuonProducer::build_tkMuons_from_idxs(TkMuonCollection& tkMuons,
-                                               const std::vector<int>& matches,
-                                               const edm::Handle<L1TTTrackCollectionType>& l1tksH,
-                                               int detector) const {
-  for (uint imatch = 0; imatch < matches.size(); ++imatch) {
-    int match_trk_idx = matches.at(imatch);
```

Comment:
There were two functions that only differed by one line. Since a default-constructed `edm::Handle` is invalid, I unified the functions by checking if the corresponding argument is valid (a few lines below this). This reduces code duplication.